target_chains/cosmwasm/tools/src/chains-manager/chain-querier.ts

```diff
@@ -0,0 +1,102 @@
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { QueryClient } from "@cosmjs/stargate";
+import { WasmExtension, setupWasmExtension } from "@cosmjs/cosmwasm-stargate";
+
+export type ContractInfoRequest = {
+  contractAddr: string;
+};
+
+export type ContractInfoResponse = {
+  codeId: number;
+  creator: string;
+  admin: string;
+  label: string;
+};
+
+export type SmartContractRequest = {
+  contractAddr: string;
+  query: Object;
+};
+
+export type RawContractStateRequest = {
+  contractAddr: string;
+  key: Buffer;
+};
+
+export type AllContractStateRequest = {
+  contractAddr: string;
+};
+
+export interface ChainQuerier {
+  getContractInfo(req: ContractInfoRequest): Promise<ContractInfoResponse>;
+  getSmartContractState(req: SmartContractRequest): Promise<Object>;
+  getRawContractState(req: RawContractStateRequest): Promise<Object>;
+  getAllContractState(req: AllContractStateRequest): Promise<Object>;
+}
+
+export class GenericQuerier implements ChainQuerier {
```

Comment:
a doc comment at the top of this class would be helpful too, e.g., to explain why this is a "generic" querier